controllers/topology/internal/extensions/patches/inline/json_patch_generator.go

```diff
@@ -347,7 +347,7 @@ func parsePathSegment(segment string) (*pathSegment, error) {
 	}
 
 	return &pathSegment{
-		path:  segment[:strings.Index(segment, leftArrayDelim)], //nolint:gocritic // We already check above that segment contains leftArrayDelim,
+		path:  segment[:strings.Index(segment, leftArrayDelim)], //nolint:gocritic
```

Comment:
Can we keep the comment?

Reply:
Honestly can't remember why this was dropped in the first place. :D